### ISSUE TITLE
[css-color-6] Make color-contrast() candidates optional

### ISSUE THREAD

**Issue by LeaVerou** [w3c/csswg-drafts#7345]
From https://www.w3.org/TR/css-color-5/#winner

> If there is a target contrast, and the end of the list is reached without meeting that target, either [white](https://www.w3.org/TR/css-color-4/#valdef-color-white) or [black](https://www.w3.org/TR/css-color-4/#valdef-color-black) is returned, whichever has the higher contrast.

However, a lot of the time getting either `black` or `white` is *exactly* what you want. If this is the default behavior, people shouldn't have to do dummy things like `color-contrast(var(--color) vs white, black)` or even `color-contrast(var(--color) vs var(--color), var(--color))` etc, they should just be able to do something like `color-contrast(var(--color))` and get either white or black.

I.e. change the grammar from:

```
color-contrast() = color-contrast( <color> vs <color>#{2,}  [ to [<number> | AA | AA-large | AAA | AAA-large]]? )
```

to:

```
color-contrast() = color-contrast( <color> [ vs <color>#{2,}  [ to [<number> | AA | AA-large | AAA | AAA-large]]? ]? )
```

**Comment by svgeesus** [comment]
Adding the `white`|`black` came after target contrast which came after the initial design and grammar.

So yes, lets update the grammar.

Also, we don't say what is returned at the exact luminance where `white` and `black` have the same contrast.

**Comment by LeaVerou** [comment]
Agenda+ so we can resolve soon since this is part of Interop 2022.

**Comment by manuelmeister** [comment]
Why do we use `vs` anyway and not just something like this:

```
color-contrast() = color-contrast( <color> [ , <color>#{2,}  [ , to [<number> | AA | AA-large | AAA | AAA-large]]? ]? )
```

Because for me, the following looks like only maroon is tested to AA and only wheat and bisque are compared.
```
color-contrast(wheat vs bisque, darkgoldenrod, olive, sienna, darkgreen, maroon to AA)
```

I think this would be much more readable:
```
color-contrast(wheat, bisque darkgoldenrod olive sienna darkgreen maroon, to AA)
```

**Comment by svgeesus** [comment]
Why the 2 in the grammar: `[ vs <color>#{2,}`?

`color-contrast(wheat)` gives white or black
`color-contrast(wheat vs maroon)` gives maroon or white or black
`color-contrast(wheat vs tan, maroon)` givestan or maroon or white or black

**Comment by svgeesus** [comment]
So in other words, if there is a target contrast, the minimum list length is zero, and a list of one makes sense.
If there is no target contrast, the minimum list length of 2 continues to make sense (a list of one returns that one, always).

**Comment by LeaVerou** [comment]
Closed via https://github.com/w3c/csswg-drafts/commit/f5a104b8078183bcd09ba866231ea31766f1fe50

**Comment by svgeesus** [comment]
"The candidate colors are tested sequentially, from left to right"

If there are no explicit candidates, are the candidate colors white, black or black, white? Usually one will be the clear winner but there will be a particular grey level where both give the same (poor) contrast and we want implementations to reliably interop for that color.

@LeaVerou
